src/napari_matplotlib/scatter.py

```diff
@@ -222,6 +217,13 @@ def on_update_layers(self) -> None:
         """
         Called when the layer selection changes by ``self.update_layers()``.
         """
+        if hasattr(self, "_key_selection_widget"):
+            self._key_selection_widget.reset_choices()
+
+        # reset the axis keys
+        self._x_axis_key = None
+        self._y_axis_key = None
```

Comment:
This should go in the new class, which can call `super().on_update_layers()` as part of it's call.

Reply:
@dstansby I'm not sure I got that right - I basically copied line 220 and 221 from the `on_update_layers` method of the `FeaturesScatterWidget`. Is that what you meant?

26a4057286cd2f234e92da4fbc7eebf7b388f249

Reply:
> Sorry, think I got a bit confused here, ignore my original comment 

I think I got confused as well and accidentally modified the wrong file 🙈 - will roll back momentarily

Reply:
😆